File: cattus/tests/Lang/Python/testdata/test_ConstructClass_aaaa_bbbb_cccc_entity_builder.generate.py

class aaaa_bbbb_cccc_entity_builder():

    __eeee_ffff_gggg: int

    __iiii_jjjj_kkkk: int

    __mmmm_nnnn_oooo: int

    @property
    def eeee_ffff_gggg(self) -> int:
        return self.__eeee_ffff_gggg

    @property
    def iiii_jjjj_kkkk(self) -> int:
        return self.__iiii_jjjj_kkkk

    @property
    def mmmm_nnnn_oooo(self) -> int:
        return self.__mmmm_nnnn_oooo

    def __init__(self, eeee_ffff_gggg: int, iiii_jjjj_kkkk: int, mmmm_nnnn_oooo: int) -> None:
        self.__eeee_ffff_gggg = eeee_ffff_gggg
        self.__iiii_jjjj_kkkk = iiii_jjjj_kkkk
        self.__mmmm_nnnn_oooo = mmmm_nnnn_oooo

    def __init__(self, eeee_ffff_gggg: int=1, iiii_jjjj_kkkk: int=2, mmmm_nnnn_oooo: int=3) -> None:
        self.__eeee_ffff_gggg = eeee_ffff_gggg
        self.__iiii_jjjj_kkkk = iiii_jjjj_kkkk
        self.__mmmm_nnnn_oooo = mmmm_nnnn_oooo

    def __init__(self, that: aaaa_bbbb_cccc_entity) -> None:
        self.__eeee_ffff_gggg = that.eeee_ffff_gggg
        self.__iiii_jjjj_kkkk = that.iiii_jjjj_kkkk
        self.__mmmm_nnnn_oooo = that.mmmm_nnnn_oooo

    def build(self) -> aaaa_bbbb_cccc_entity:
        pass

File: cattus/tests/Lang/CSharp/testdata/test_ConstructClass_AaaaBbbbCcccEntityBuilder.generate.cs
namespace AaaaBbbbCccc {

    public class AaaaBbbbCcccEntityBuilder {

        private sbyte eeeeFfffGggg;

        private short iiiiJjjjKkkk;

        private int mmmmNnnnOooo;

        public sbyte EeeeFfffGggg() {
            return this.eeeeFfffGggg;
        }

        public short IiiiJjjjKkkk() {
            return this.iiiiJjjjKkkk;
        }

        public int MmmmNnnnOooo() {
            return this.mmmmNnnnOooo;
        }

        public AaaaBbbbCcccEntityBuilder(sbyte eeeeFfffGggg, short iiiiJjjjKkkk, int mmmmNnnnOooo) {
            this.eeeeFfffGggg = eeeeFfffGggg;
            this.iiiiJjjjKkkk = iiiiJjjjKkkk;
            this.mmmmNnnnOooo = mmmmNnnnOooo;
        }

        public AaaaBbbbCcccEntityBuilder(sbyte eeeeFfffGggg=1, short iiiiJjjjKkkk=2, int mmmmNnnnOooo=3) {
            this.eeeeFfffGggg = eeeeFfffGggg;
            this.iiiiJjjjKkkk = iiiiJjjjKkkk;
            this.mmmmNnnnOooo = mmmmNnnnOooo;
        }

        public AaaaBbbbCcccEntityBuilder(AaaaBbbbCcccEntity that) {
            this.eeeeFfffGggg = that.EeeeFfffGggg;
            this.iiiiJjjjKkkk = that.IiiiJjjjKkkk;
            this.mmmmNnnnOooo = that.MmmmNnnnOooo;
        }

        public AaaaBbbbCcccEntity Build() {
        }

    }

}
